### PR TITLE
K8SPXC-252 Add operator user to proxysql

### PR DESCRIPTION
**Pull request by nonemax**
[K8SPXC-252](https://jira.percona.com/browse/K8SPXC-252)

### CODE REVIEW

File: pkg/pxc/app/statefulset/proxysql.go [hunk 1]
```diff
@@ -82,12 +82,6 @@ func (c *Proxy) AppContainer(spec *api.PodSpec, secrets string, cr *api.PerconaX
 				Name:  "PXC_SERVICE",
 				Value: c.labels["app.kubernetes.io/instance"] + "-pxc",
 			},
-			{
```
Comment (delgod) [inline pkg/pxc/app/statefulset/proxysql.go:85]: remove and add vars only during the update to 1.5.0

Comment (nonemax) [reply pkg/pxc/app/statefulset/proxysql.go:85]: https://github.com/percona/percona-xtradb-cluster-operator/pull/452/commits/83ec682d736435e54368f4030320936ce2644520

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | failed |
| big-data | failed |
| recreate | failed |
| storage | failed |
| init-deploy | failed |
| self-healing | failed |
| self-healing-advanced | failed |
| limits | failed |
| monitoring | failed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | failed |
| monitoring-2-0 | failed |
| demand-backup | failed |
| upgrade | failed |
| affinity | failed |
| operator-self-healing | passed |
| upgrade-consistency | failed |
| security-context | failed |
| scheduled-backup | failed |
| one-pod | failed |
| demand-backup-encrypted-with-tls | failed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | failed |
| init-deploy | failed |
| big-data | passed |
| recreate | failed |
| storage | failed |
| scaling-proxysql | failed |
| self-healing | passed |
| limits | failed |
| monitoring | failed |
| monitoring-2-0 | failed |
| restore-to-encrypted-cluster | failed |
| affinity | failed |
| upgrade | failed |
| one-pod | passed |
| upgrade-consistency | failed |
| security-context | failed |
| self-healing-advanced | passed |
| auto-tuning | failed |
| demand-backup | passed |
| operator-self-healing | failed |
| proxysql-sidecar-res-limits | failed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | failed |

File: pkg/pxc/app/statefulset/proxysql.go (hunk 1 above)
Comment (delgod) [reply pkg/pxc/app/statefulset/proxysql.go:85]: please don't delete MYSQL_ROOT_PASSWORD var if version <= 1.4.0 😞

File: pkg/pxc/app/statefulset/proxysql.go [hunk 2]
```diff
@@ -108,6 +102,24 @@ func (c *Proxy) AppContainer(spec *api.PodSpec, secrets string, cr *api.PerconaX
 		SecurityContext: spec.ContainerSecurityContext,
 	}
 
+	if cr.CompareVersionWith("1.5.0") < 0 {
+		appc.Env = append(appc.Env, corev1.EnvVar{
+			Name: "MYSQL_ROOT_PASSWORD",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: app.SecretKeySelector(secrets, "root"),
+			},
+		})
```
Comment (delgod) [inline pkg/pxc/app/statefulset/proxysql.go:111]: upgrade consistency test become broken because this line add MYSQL_ROOT_PASSWORD var to the end of array

Comment (nonemax) [reply pkg/pxc/app/statefulset/proxysql.go]: https://github.com/percona/percona-xtradb-cluster-operator/pull/452/commits/59f55b53163b0d98ff1a0a21838cf56dfa778ceb

https://github.com/percona/percona-xtradb-cluster-operator/pull/452/commits/319f19d5c01565df826489485f9c0d21cb201b1b

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| scaling | passed |
| init-deploy | failed |
| big-data | passed |
| storage | failed |
| self-healing | passed |
| limits | failed |
| monitoring | failed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | failed |
| monitoring-2-0 | failed |
| self-healing-advanced | passed |
| affinity | failed |
| one-pod | passed |
| upgrade-consistency | passed |
| auto-tuning | failed |
| security-context | failed |
| demand-backup | passed |
| operator-self-healing | passed |
| proxysql-sidecar-res-limits | failed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | failed |
| big-data | failed |
| recreate | passed |
| self-healing | failed |
| scaling | passed |
| limits | passed |
| monitoring | failed |
| scaling-proxysql | passed |
| monitoring-2-0 | passed |
| self-healing-advanced | failed |
| restore-to-encrypted-cluster | failed |
| upgrade | failed |
| demand-backup | failed |
| affinity | passed |
| operator-self-healing | failed |
| scheduled-backup | failed |
| one-pod | passed |
| auto-tuning | failed |
| demand-backup-encrypted-with-tls | failed |
| proxysql-sidecar-res-limits | passed |
| upgrade-consistency | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| recreate | failed |
| big-data | failed |
| init-deploy | failed |
| scaling | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| limits | passed |
| upgrade | failed |
| restore-to-encrypted-cluster | failed |
| self-healing-advanced | failed |
| monitoring | passed |
| operator-self-healing | passed |
| demand-backup | failed |
| scheduled-backup | failed |
| monitoring-2-0 | failed |
| affinity | failed |
| upgrade-consistency | passed |
| one-pod | failed |
| demand-backup-encrypted-with-tls | failed |
| security-context | failed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | failed |
| scaling | passed |
| init-deploy | failed |
| restore-to-encrypted-cluster | failed |
| scaling-proxysql | passed |
| demand-backup | failed |
| scheduled-backup | failed |
| limits | failed |
| upgrade | passed |
| monitoring | failed |
| upgrade-consistency | passed |
| security-context | failed |
| demand-backup-encrypted-with-tls | failed |
| monitoring-2-0 | failed |
| affinity | failed |
| one-pod | failed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| init-deploy | passed |
| big-data | failed |
| recreate | passed |
| storage | passed |
| scaling-proxysql | passed |
| limits | passed |
| upgrade | failed |
| self-healing | failed |
| restore-to-encrypted-cluster | failed |
| demand-backup | failed |
| self-healing-advanced | passed |
| monitoring | failed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| security-context | failed |
| operator-self-healing | passed |
| affinity | passed |
| scheduled-backup | failed |
| one-pod | passed |
| auto-tuning | passed |
| demand-backup-encrypted-with-tls | failed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| scaling | passed |
| recreate | passed |
| big-data | passed |
| init-deploy | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| limits | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | passed |
| monitoring | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| security-context | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| affinity | passed |
| scheduled-backup | passed |
| one-pod | passed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| storage | passed |
| big-data | passed |
| scaling | passed |
| recreate | passed |
| scaling-proxysql | failed |
| self-healing | passed |
| limits | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | passed |
| monitoring | failed |
| upgrade-consistency | passed |
| self-healing-advanced | failed |
| monitoring-2-0 | passed |
| security-context | passed |
| demand-backup | failed |
| affinity | passed |
| operator-self-healing | passed |
| scheduled-backup | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | passed |

**Pull request merged**